File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/test/TestSwlCertExchanger.java
package i2f.swl.test;

import i2f.swl.cert.SwlCertExchanger;
import i2f.swl.cert.data.SwlCert;
import i2f.swl.cert.impl.SwlResourceCertManager;
import i2f.swl.data.SwlData;
import i2f.swl.impl.SwlAesSymmetricEncryptor;
import i2f.swl.impl.SwlBase64Obfuscator;
import i2f.swl.impl.SwlRsaAsymmetricEncryptor;
import i2f.swl.impl.SwlSha256MessageDigester;

import java.util.Arrays;

/**
 * @author Ice2Faith
 * @date 2024/7/10 10:12
 * @desc
 */
public class TestSwlCertExchanger {
    public static void main(String[] args) throws Exception {
//        testRaw();

        testCert();
    }

    public static void testCert() throws Exception {
        SwlCertExchanger clientTransfer = new SwlCertExchanger();
        clientTransfer.setAsymmetricEncryptorSupplier(() -> new SwlRsaAsymmetricEncryptor());
        clientTransfer.setSymmetricEncryptorSupplier(() -> new SwlAesSymmetricEncryptor());
        clientTransfer.setMessageDigester(new SwlSha256MessageDigester());
        clientTransfer.setObfuscator(new SwlBase64Obfuscator());
        clientTransfer.setCertManager(new SwlResourceCertManager());

        SwlCertExchanger serverTransfer = new SwlCertExchanger();
        serverTransfer.setAsymmetricEncryptorSupplier(() -> new SwlRsaAsymmetricEncryptor());
        serverTransfer.setSymmetricEncryptorSupplier(() -> new SwlAesSymmetricEncryptor());
        serverTransfer.setMessageDigester(new SwlSha256MessageDigester());
        serverTransfer.setObfuscator(new SwlBase64Obfuscator());
        serverTransfer.setCertManager(new SwlResourceCertManager());

        String certId = "test";
        serverTransfer.createCertPair(certId);
        SwlCert clientCert = serverTransfer.getCertManager().loadClient(certId);
        clientTransfer.getCertManager().storeClient(clientCert);


        SwlData clientSendData = clientTransfer.sendByCertId(certId,
                Arrays.asList("body:123456", "query:user=admin"));


        String clientId = "127.0.0.1";
        SwlData serverReceiveData = serverTransfer.acceptByCertId(
                clientSendData,
                certId,
                clientId);

        SwlData serverResponseData = serverTransfer.responseByCertId(certId,
                Arrays.asList("echo:ok", "data:ok"));

        String serverId = "server";
        SwlData clientReceiveData = clientTransfer.receiveByCert(
                serverResponseData,
                clientCert
        );

        System.out.println("ok");
    }

}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/test/TestSwlCertExchanger.js
import SwlRsaAsymmetricEncryptorSupplier from "../core/impl/supplier/SwlRsaAsymmetricEncryptorSupplier";
import SwlAesSymmetricEncryptorSupplier from "../core/impl/supplier/SwlAesSymmetricEncryptorSupplier";
import SwlSha256MessageDigesterSupplier from "../core/impl/supplier/SwlSha256MessageDigesterSupplier";
import SwlBase64Obfuscator from "../core/impl/SwlBase64Obfuscator";
import SwlCertExchanger from "../core/cert/SwlCertExchanger";
import SwlMemCertManager from "../core/cert/impl/SwlMemCertManager";

/**
 * @return TestSwlCertExchanger
 * @constructor
 */
function TestSwlCertExchanger() {

}

TestSwlCertExchanger.main=function(){
    TestSwlCertExchanger.testCert()
}

TestSwlCertExchanger.testCert = function () {

    let clientTransfer = new SwlCertExchanger();
    clientTransfer.asymmetricEncryptorSupplier = (new SwlRsaAsymmetricEncryptorSupplier());
    clientTransfer.symmetricEncryptorSupplier = (new SwlAesSymmetricEncryptorSupplier());
    clientTransfer.messageDigesterSupplier = (new SwlSha256MessageDigesterSupplier());
    clientTransfer.obfuscator = (new SwlBase64Obfuscator());
    clientTransfer.certManager=(new SwlMemCertManager())

    let serverTransfer = new SwlCertExchanger();
    serverTransfer.asymmetricEncryptorSupplier = (new SwlRsaAsymmetricEncryptorSupplier());
    serverTransfer.symmetricEncryptorSupplier = (new SwlAesSymmetricEncryptorSupplier());
    serverTransfer.messageDigesterSupplier = (new SwlSha256MessageDigesterSupplier());
    serverTransfer.obfuscator = (new SwlBase64Obfuscator());
    serverTransfer.certManager=(new SwlMemCertManager())


    let certId = "test";
    serverTransfer.createCertPair(certId);
    let clientCert = serverTransfer.certManager.loadClient(certId);
    clientTransfer.certManager.storeClient(clientCert);


    let clientSendData = clientTransfer.sendByCertId(certId,
        ["body:123456", "query:user=admin"]);


    let clientId = "127.0.0.1";
    let serverReceiveData = serverTransfer.acceptByCertId(
        clientSendData,
        certId,
        clientId);

    let serverResponseData = serverTransfer.responseByCertId(certId,
        ["echo:ok", "data:ok"]);

    let serverId = "server";
    let clientReceiveData = clientTransfer.receiveByCert(
        serverResponseData,
        clientCert
    );

    console.log("ok");
}

export default TestSwlCertExchanger
